### PR TITLE
Add python_requires to setup().

### LINKED ISSUE #1005: Packaging: Add python_requires to manifest
Define our Python version requirements in our package manifest, as described here: https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires

### PR DESCRIPTION
**Pull request by cmarqu**
Closes #1005.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ def package_files(directory):
     author='Chris Higgs, Stuart Hodgson',
     author_email='cocotb@potentialventures.com',
     install_requires=[],
+    python_requires='=2.7, >=3.5',
```
Comment (imphil) [inline setup.py]: =2.7 should be ==2.7, see https://www.python.org/dev/peps/pep-0440/#examples

**Comment by cmarqu** [comment]
Ugh, sorry for the many commits... Not even the simplest changes should be done on a cell phone.

**Comment by imphil** [comment]
What's the problem with `==2.7, >=3.5`?

**Comment by cmarqu** [comment]
> What's the problem with `==2.7, >=3.5`?

It did not work (https://travis-ci.com/cmarqu/cocotb/builds/131656099). For instance, ``==2.7`` means 2.7 and 2.7.0, but not 2.7.15.

**Comment by imphil** [comment]
Use `~=2.7` then, at least thats what the docs suggest. And please rebase into to a single commit.

**Comment by cmarqu** [comment]
``~=2.7, ...`` did not work (see https://travis-ci.com/cmarqu/cocotb/builds/131711726), so I reverted back to ``>=``. 
The specifier format feels intuitive at first since the comma means **and**.

**Comment by imphil** [comment]
Ah OK, the comma behavior is slightly unexpected. Thanks for trying!

**Pull request merged**